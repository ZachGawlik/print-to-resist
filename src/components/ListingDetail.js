import React, { PropTypes } from 'react';
import ListingTile from './ListingTile';
import { BASE_URL } from '../constants';
import '../styles/ListingDetail.css';

const ListingDetail = ({
  isPrintInputOpen,
  listing,
  onCopiesChange,
  onPrintingSubmit,
  openPrintInput
}) => (
  <div className="ListingDetail">
    <div className="ListingDetail__info">
      <ListingTile listing={listing} />
      <div className="ListingDetail__description">
        <h3>Description:</h3>
        <p>{listing.description}</p>
      </div>
    </div>
    <div className="ListingDetail__files">
      <h3 className="ListingDetail__files-header">Poster files:</h3>
      <ul>
      {listing.imageIds.map((imageId, index) => (
        <li className="ListingDetail__files-link" key={imageId}>
          <a href={`${BASE_URL}/image/${imageId}`} target="_blank">
            Poster {index + 1}
          </a>
        </li>
      ))}
      </ul>
    </div>
    <div>
      <h3>Ready to print?</h3>
      <p>
        Open up the poster's file(s) above, and print them as you would anything else.
      </p>
      <p>
        Remember to print them in <strong>{listing.isColor ? 'color' : 'black and white'}</strong>,
        using <strong>{listing.paperSize}</strong> sized paper.
      </p>
      <p>
        Once printed, take your posters and stroll around town!
        If it's a traditional poster, grab a roll of tape.
        If it's more suited for takeaway informational flyers, leave them in key places.
      </p>
      {listing.instruction &&
        <div>
          <p>
            The person who uploaded the poster left some specific guidance for how to distribute the poster:
          </p>
          <p>
            {listing.instruction}
          </p>
        </div>
      }
      {isPrintInputOpen ?
        <div>
          <p>Let them know the word is being spread</p>
          <label>
            Copies:
            <input type="number" min="1" max="200" onChange={onCopiesChange} />
          </label>
          <button onClick={onPrintingSubmit}>Submit</button>
        </div> :
        <button onClick={openPrintInput}>Done!</button>
      }
    </div>
  </div>
);

ListingDetail.propTypes = {
  isPrintInputOpen: PropTypes.bool,
  listing: PropTypes.object.isRequired,
  onCopiesChange: PropTypes.func.isRequired,
  onPrintingSubmit: PropTypes.func.isRequired,
  openPrintInput: PropTypes.func.isRequired
};

export default ListingDetail
