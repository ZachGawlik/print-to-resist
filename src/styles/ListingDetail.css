.ListingDetail__tile {
  width: 300px;
}

.ListingDetail__files {
  margin: 5em 0;
}

.ListingDetail__files-header {
  margin-bottom: 0;
}

@media (min-width: 600px) {
  .ListingDetail__info {
    display: flex;
  }

  .ListingDetail__description {
    margin-left: 20px;
  }
}
