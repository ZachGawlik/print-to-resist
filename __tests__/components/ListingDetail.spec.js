import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ListingDetail from '../../src/components/ListingDetail';

const listing97 = {
  listingId: 97,
  deadline: '2017-06-01T04:00:00.000Z',
  description: 'Upcoming dates for NYC congress town halls',
  instruction: '',
  isColor: false,
  paperSize: 'Letter',
  dateSubmitted: '2017-04-13T13:49:47.000Z',
  isApproved: 1,
  title: 'NYC Congress Town Halls',
  thumbnailId: 'c0c51fba8b26122d2c368abbfd5da8b6.png',
  imageIds: ['c0753ac7045056c7f73779a80db8ae55.png'],
  tags: ['nyc', 'town hall'],
  links: []
};

describe('ListingDetail', () => {
  it('renders as expected', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <ListingDetail
        isPrintInputOpen={true}
        listing={listing97}
        onCopiesChange={jest.fn()}
        onPrintingSubmit={jest.fn()}
        openPrintInput={jest.fn()}
      />
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
