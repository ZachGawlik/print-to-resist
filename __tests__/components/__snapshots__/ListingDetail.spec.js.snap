// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetail renders as expected 1`] = `
<div
  className="ListingDetail"
>
  <div
    className="ListingDetail__info"
  >
    <div
      className="ListingDetail__tile"
    >
      <ListingTile
        deadline="2017-06-01T04:00:00.000Z"
        tags={
          Array [
            "nyc",
            "town hall",
          ]
        }
        thumbnailId="c0c51fba8b26122d2c368abbfd5da8b6.png"
        title="NYC Congress Town Halls"
      />
    </div>
    <div
      className="ListingDetail__description"
    >
      <h3>
        Description:
      </h3>
      <p>
        Upcoming dates for NYC congress town halls
      </p>
    </div>
  </div>
  <div
    className="ListingDetail__files"
  >
    <h3
      className="ListingDetail__files-header"
    >
      Poster files:
    </h3>
    <ul>
      <li
        className="ListingDetail__files-link"
      >
        <a
          href="http://localhost:3000/image/c0753ac7045056c7f73779a80db8ae55.png"
          target="_blank"
        >
          Poster 
          1
        </a>
      </li>
    </ul>
  </div>
  <div>
    <h3>
      Ready to print?
    </h3>
    <p>
      Open up the poster's file(s) above, and print them as you would anything else.
    </p>
    <p>
      Remember to print them in
       
      <strong>
        black and white
      </strong>
      , using 
      <strong>
        Letter
      </strong>
       sized paper.
    </p>
    <p>
      Once printed, take your posters and stroll around town! If it's a traditional poster, grab a roll of tape. If it's more suited for takeaway informational flyers, leave them in key places.
    </p>
    
    <div>
      <p>
        Let them know the word is being spread
      </p>
      <label>
        Copies:
        <input
          max="200"
          min="1"
          onChange={[Function]}
          type="number"
        />
      </label>
      <button
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;
